pkg/apis/core/validation/validation.go

```diff
@@ -1201,6 +1207,9 @@ func validateRBDPersistentVolumeSource(rbd *core.RBDPersistentVolumeSource, fldP
 	if len(rbd.RBDImage) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("image"), ""))
 	}
+	if rbd.Keyring != "" && !filePathRegex.MatchString(rbd.Keyring) {
```

Comment:
this is an existing field... can't tighten validation on it now

Reply:
@rootfs Asked for validation of the API field because there should have been a validation in the first place. Ist this something you guys want to discuss? I'm fine with either way and can move the validation to the provisioner.

FWIW: I think API field validation is the better way as it is more clear for the user. Storage provisioner errors only show up in the master logs.

Reply:
@liggitt validating keyring is something should have been done but not addressed (fault is mine).

Reply:
> there should have been a validation in the first place. Ist this something you guys want to discuss? I'm fine with either way and can move the validation to the provisioner.
> 
> FWIW: I think API field validation is the better way as it is more clear for the user

I don't disagree, but tightening validation on existing data can break upgrade/migration and should only been done when absolutely required. 

> @liggitt validating keyring is something should have been done but not addressed

Understood, but on a shipped field it is generally too late

Reply:
@liggitt I can fully understand why you wouldn't want to change the behaviour of an existing API field. The validation however is not very strict - any existing setup which would violate the field validation wouldn't work in the first place. So introducing the validation should not have any effect on clusters whith a working configuration. On top of this, the validation is much easier to test when done in the API.

If I understand you correctly though, changing stable API fields is simply not up for discussion for policy reasons. If this is the case and @rootfs agrees, I'll move the validation to the provisioner.

Reply:
> If I understand you correctly though, changing stable API fields is simply not up for discussion for policy reasons

In general, that's the policy. Tightening validation takes a local problem (a particular volume wouldn't work) and makes it a global problem on an upgrade (migration fails, and the offending object cannot be deleted). There are exceptions to tightening existing validation, but the bar is generally higher than "we want to fail earlier for usability"

Reply:
One could argue that using an unvalidated keyring path with `exec.Run()` is potentially insecure. PVs should only be managed by cluster admins, but still.

@rootfs If this is fine for you, too, I'll move the validation to the provisioner.

Reply:
@madddi ok, let's defer validation to rbd volume. Thanks